library/core/src/ptr/non_null.rs

```diff
@@ -1770,3 +1779,30 @@ impl<T: ?Sized> From<&T> for NonNull<T> {
         unsafe { NonNull { pointer: reference as *const T } }
     }
 }
+
+//#[unstable(feature="kani", issue="none")]
+mod verify {
+    use super::*;
+
+    // pub const unsafe fn new_unchecked(ptr: *mut T) -> Self
+    #[kani::proof_for_contract(NonNull::new_unchecked)]
+    pub fn non_null_check_new_unchecked() {
+        let mut x : i32 = kani::any();
+        let xptr = &mut x;
+        unsafe {
+            let _ = NonNull::new_unchecked(xptr as *mut i32);
```

Comment:
You can also pass a potentially NULL pointer here, and it will be ruled out by the `requires` clause, which will assume that the pointer is not null.

Reply:
It's actually the opposite, we want the harness to allow all possible value combinations for the inputs to ensure the contract rules out all cases that could trigger UB.